lib/koans/19_genservers.ex

```diff
@@ -0,0 +1,153 @@
+     {:reply, current_password, current_password} 
+    end
+
+    def handle_call(:get_manufacturer, _from, current_state) do
+      {:reply, "Apple Inc.", current_state}
+    end
+
+    def handle_call(:get_type, _from, current_state) do
+      {:reply, "MacBook Pro", current_state}
+    end
+
+    def handle_call(:get_owner_name, _from, current_state) do
+      {:reply, {:ok, "Jack Sparrow"}, current_state}
+    end
+
+    def handle_call(:get_specs, _from, current_state) do
+      {:reply, {:ok, ["2.9 GHz Intel Core i5"], 8192, :intel_iris_graphics}, current_state}
+    end
+
+    def handle_call(:name_check, _from, current_state) do
+      {:reply, "Congrats! Your process was successfully named.", current_state}
+    end
+    
+    def handle_call({:unlock, password}, _from, current_password) do
+      case password do
+        password when password === current_password ->
+          {:reply, {:ok, "Laptop unlocked!"}, current_password}
+        _ -> 
+          {:reply, {:error, "Incorrect password!"}, current_password}
+      end
+    end
+    
+    def handle_cast({:change_password, old_password, new_password}, current_password) do
+      case old_password do
+        old_password when old_password == current_password ->
+          {:noreply, new_password}
+        _ ->
+          {:noreply, current_password} 
+      end
+    end
+  end
+
+  koan "Servers that are created and initialized successfully returns a tuple that holds the PID of the server" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    assert is_pid(pid) == ___
+  end
+
+  koan "When starting a GenServer you can set it's initial state" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    assert GenServer.call(pid, :get_password) == ___
+  end
+
+  koan "The handle_call callback is synchronous so it will block until a reply is received" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    assert GenServer.call(pid, :get_password) == ___
+  end
+
+  koan "A server can support multiple actions by implementing multiple handle_call functions" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    assert GenServer.call(pid, :get_manufacturer) == ___
+    assert GenServer.call(pid, :get_type) == ___
+  end
+
+  koan "A handler can return multiple values and of different types" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    {:ok, processor, memory, graphics} = GenServer.call(pid, :get_specs)
+    assert processor == ___
+    assert memory == ___
+    assert graphics == ___
+  end
+
+  koan "The handle_cast callback handles asynchronous messages" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    GenServer.cast(pid, {:change_password, "3kr3t!", "Hello"})
+    assert GenServer.call(pid, :get_password) == ___
+  end
+
+  koan "Handlers can also return error responses" do
+    {:ok, pid} = GenServer.start_link(Laptop, "3kr3t!")
+    assert GenServer.call(pid, {:unlock, 2017}) == ___
```

Comment:
Is it confusing that the attempted password (`2017`) looks like a year or something? Maybe it should be the same type and resemble a password?